Added comments to platform messages

KipHamiltons opened the pull request:
Let me know if I should add a different reviewer.
Most of this is probably wrong of course.

Bidski commented on shared/message/platform/darwin/DarwinSensors.proto:
```diff
@@ -22,13 +22,16 @@ package message.platform.darwin;
 message DarwinSensors {
-    // bitmask values
+
+    /**
+     * Set of error flags/statuses.
+     * bitmask values.
+     */
```
```suggestion
    /**
     * Set of error flags/statuses.
     * bitmask values.
     * See https://emanual.robotis.com/docs/en/dxl/protocol1/#protocol
     */
```

Bidski commented on shared/message/platform/darwin/DarwinSensors.proto:
```diff
@@ -85,27 +162,113 @@ message DarwinSensors {
+         */
+        float goalPosition = 6;
+
+        float movingSpeed = 7;
```
What is moving speed???

TrentHouliston replied:
Moving speed is what dynamixel calls servo velocity

Bidski commented on shared/message/platform/darwin/DarwinSensors.proto:
```diff
@@ -129,21 +292,82 @@ message DarwinSensors {
+    /**
+     * Holds the error code (if any) from the CM740.
+     */
+    uint32 cm740ErrorFlags = 2;
```
Do we know what the possible error values/bits/flags are? Are they the same for the servos above?

TrentHouliston replied:
They are the same as for the servos mostly. Only some of the errors can't occur on the CM740

JosephusPaye replied:
Can we get a link to a reference with possible error codes?

ysims commented on shared/message/platform/gazebo/Command.proto:
```diff
@@ -20,12 +20,16 @@ syntax = "proto3";
+ */
 message Command {
+
+    /// RESET - reset entire simulation; RESET_TIME - Reset simulation time and realtime clocks back to 0
```
I feel like this line is a bit redundant

ysims commented on shared/message/platform/darwin/DarwinSensors.proto:
```diff
@@ -85,27 +107,63 @@ message DarwinSensors {
+        /// Derivative gain. Does not consider error, just dampens forces to reduce derivatives. Gain values are in
+        /// range 0~254. K_d : d_gain * 4 / 1000
+        float dGain = 5;
+        /// It is a position value of destination. 0 to 4,095 (0xFFF) is available. The unit is 0.088 [°].
```
```suggestion
        /// Position value of destination. 0 to 4,095 (0xFFF) is available. The unit is 0.088 degrees.
```

ysims commented on shared/message/platform/darwin/DarwinSensors.proto:
```diff
@@ -85,27 +107,63 @@ message DarwinSensors {
+        float movingSpeed = 7;
+        /// Used to set the servo on or off. Typically either 0 (off) or 100 (on)
+        float torque = 8;
+        /// The range of the value is 0~4095 (0xFFF), and the unit is 0.088 [°]
```
```suggestion
        /// The range of the value is 0~4095 (0xFFF), and the unit is 0.088 degrees
```

ysims commented on shared/message/platform/darwin/DarwinSensors.proto:
```diff
@@ -128,22 +186,51 @@ message DarwinSensors {
+
+/**
+ * Empty message to indicate button press
+ */
```
Would be good to get some info on what up/down mean. From what I understand, up is when it's been pressed, and down is when it's just stopped being pressed?

ysims commented on shared/message/platform/darwin/DarwinSensors.proto:
```diff
@@ -128,22 +186,51 @@ message DarwinSensors {
+    Buttons buttons = 6;
+    /// Voltage for the whole robot, from the main battery
+    float voltage = 7;
+    /// The value is between 0 and 1023, representing a value between -4g and 4g => 512 = 0g.
```
This info would be better put in above in the Accelerometer message definition

ysims commented on shared/message/platform/gazebo/Command.proto:
```diff
@@ -20,12 +20,17 @@ syntax = "proto3";
 package message.platform.gazebo;
 
+/**
+ * Command to reset time, or entire simulation
```
Probably worth mentioning this is a Gazebo thing

ysims commented on shared/message/platform/gazebo/RawSensors.proto:
```diff
@@ -22,9 +22,13 @@ package message.platform.gazebo;
 import "message/platform/darwin/DarwinSensors.proto";
 
+/**
+ * A set of DarwinSensors with an associated model
```
Probably worth mentioning this is a Gazebo thing

KipHamiltons commented on shared/message/platform/darwin/DarwinSensors.proto:
```diff
@@ -42,39 +44,59 @@ message DarwinSensors {
+        /// led3 toggle status
         bool led3 = 2;
+        /// led4 toggle status
         bool led4 = 3;
```
I have removed those comments for now

ysims reviewed (dismissed):
LGTM

ysims commented on shared/message/platform/gazebo/RawSensors.proto:
```diff
@@ -23,8 +23,6 @@ package message.platform.gazebo;
-    // The name of the model in the simulation that these sensors are from
-    string model = 1;
-    // The sensors object for this model
+    string                        model   = 1;
```
I think these comments should be added back in, ppl might not know what 'model' is otherwise. I wasn't sure what it meant.

ysims commented on shared/message/platform/gazebo/ServoTargets.proto:
```diff
@@ -24,10 +24,8 @@ import "google/protobuf/timestamp.proto";
 
 message ServoTargets {
-    // The name of the model in the simulation that these servo targets are for
     string model = 1;
```
Same with this.

Bidski commented on shared/message/platform/darwin/DarwinSensors.proto:
```diff
@@ -44,115 +38,40 @@ message DarwinSensors {
-    /**
-     * The raw gyro values from the CM740 (they are read from the board in ZYX): x, y, z.
-     */
+    /// The raw gyro values from the CM740 (they are read from the board in ZYX)
```
You should add information on the range of values, same as you did for the accelerometer

KipHamiltons commented:
I realised that my comments for `DarwinSensors` were basically all wrong, because we converted to SI / our units before we input the values into the message. I have updated them as such

abihall commented on shared/message/platform/darwin/DarwinSensors.proto:
```diff
@@ -39,40 +41,37 @@ message DarwinSensors {
         bool middle = 2;
     }
-
+    /// In m/s^2. Each axis is bounded between -4g and 4g, according to the datasheet
```
What data sheet?

tayloryoung6396 replied:
https://emanual.robotis.com/docs/en/platform/op2/getting_started/

Bidski replied:
She has a point though. The datasheet should be linked in the comment

TrentHouliston reviewed (dismissed):
Don't see anything objectionable

Bidski reviewed (changes requested):
There are comments that refer to a datasheet, but no datasheet is linked. This should be fixed

KipHamiltons commented:
@Bidski could I please grab a review on this one

Bidski reviewed (approved):
:shrug:

ysims commented on shared/message/platform/gazebo/Torso.proto:
```diff
@@ -24,16 +24,15 @@ import "Matrix.proto";
 
 message Torso {
-    // The name of the model in the simulation that this position is for
     string model = 1;
```
This is the only one without `/// The name of the model in the simulation that these servo targets are for`

ysims replied:
FIX IT

ysims reviewed (approved):
Go review #672
